### ISSUE TITLE
[FG:InPlacePodVerticalScaling] kubelet.go:1875 "Unable to attach or mount volumes for pod; skipping pod"

### ISSUE THREAD

**Issue by discostur** [kubernetes/kubernetes#117652]
### What happened?

Deployed a three node HA cluster: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/

Restart all three nodes at once -> kubelet ist not able to start static control plane pods:  

```bash 
kubelet[3112]: E0427 16:27:26.004278    3112 kubelet.go:1875] "Unable to attach or mount volumes for pod; skipping pod" err="unmounted volumes=[etcd-data etcd-certs], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition" pod="kube-system/etcd-k8s-02.intern"
kubelet[3112]: E0427 16:27:26.004349    3112 kubelet.go:1875] "Unable to attach or mount volumes for pod; skipping pod" err="unmounted volumes=[k8s-certs ca-certs etc-pki], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition" pod="kube-system/kube-apiserver-k8s-02.intern"
kubelet[3112]: E0427 16:27:26.004387    3112 pod_workers.go:1281] "Error syncing pod, skipping" err="unmounted volumes=[k8s-certs ca-certs etc-pki], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition" pod="kube-system/kube-apiserver-k8s-02.intern" podUID=44025e6eb9730b94cc3154fdb554b8e3
kubelet[3112]: E0427 16:27:26.004357    3112 pod_workers.go:1281] "Error syncing pod, skipping" err="unmounted volumes=[etcd-data etcd-certs], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition" pod="kube-system/etcd-k8s-02.intern" podUID=824259c7edeeda8de0c848213e01349e
kubelet[3112]: E0427 16:27:26.006391    3112 kubelet.go:1875] "Unable to attach or mount volumes for pod; skipping pod" err="unmounted volumes=[ca-certs etc-pki flexvolume-dir k8s-certs kubeconfig], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition" pod="kube-system/kube-controller-manager-k8s-02.intern"
```

If i downgrade kubelet to `v1.26.4` all static pods start normal. The strange thing is that InPlacePodVerticalScaling is disabled by default and i didn't enable it manually. So the new feature gate shouldn't have any impact ... 

You can also see that the static "mirror" pod gets deleted first: 

```bash
kubelet[1960]: I0427 15:57:59.972629    1960 kubelet.go:1789] "Deleted mirror pod because it is outdated" pod="kube-system/kube-controller-manager-k8s-02.intern"
kubelet[1960]: I0427 15:58:00.170730    1960 kubelet.go:1789] "Deleted mirror pod because it is outdated" pod="kube-system/kube-scheduler-k8s-02.intern"
kubelet[1960]: I0427 15:58:01.564567    1960 kubelet.go:1789] "Deleted mirror pod because it is outdated" pod="kube-system/etcd-k8s-02.intern"
kubelet[1960]: I0427 15:58:01.966801    1960 kubelet.go:1789] "Deleted mirror pod because it is outdated" pod="kube-system/kube-apiserver-k8s-02.intern"
```

The error just occures if you restart all three nodes at once. If you just restart one node a time it doesn't happen. Looks like it has to do with the reachability of the apiserver. If the apiserver is available (under the loadbalancer HA IP) you can restart one node, it starts all pods and everything is fine. When you restart all nodes at once, at the start of kubelet no apiserver is available and the error occures. 

### What did you expect to happen?

Kubelet should restart all static control plane pods. 

### How can we reproduce it (as minimally and precisely as possible)?

- https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/
- deploy three node HA Cluster with stacked ETCD, external apiserver loadbalancer IP
- restart all nodes at once

### Anything else we need to know?

Kubelet full startup log: 

[kubelete_full_log.txt](https://github.com/kubernetes/kubernetes/files/11345188/kubelete_full_log.txt)

### Kubernetes version

<details>

```console
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:21:19Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:14:42Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
on prem
</details>

### OS version

<details>

```console
$ cat /etc/os-release
AlmaLinux release 8.7 (Stone Smilodon)

$ uname -a
Linux 4.18.0-425.19.2.el8_7.x86_64 #1 SMP Tue Apr 4 05:30:47 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux
```

</details>

### Install tools

<details>
kubeadm
</details>

### Container runtime (CRI) and version (if applicable)

<details>
cri-o-1.26.3-7.2.el8.x86_64
cri-tools-1.26.0-0.x86_64
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>
calico v3.25.1 
</details>

**Comment by discostur** [comment]
Did further investigate the problem - you can reproduce it also if you just restart one host (still two working / up and running hosts) and block (for example via firewall) the connection to apiserver:6443. That simulates the same situation like when you restart all nodes at once (no apiserver is available on startup yet).

**Comment by monoxane** [comment]
I am hitting this in my new cluster inconveniently after I've already started to migrate workloads, the entire thing is dead in the water and none of the nodes work.

Getting the same errors for etcd, kube-apiserver, kube-controller-manager, kube-scheduler, and kube-vip on every node, all running as static pods

kubeadm + kubelet 1.27.1

**Comment by monoxane** [comment]
downgrading with `apt install kubelet=1.26.4-00` seems to have fixed my immediate stress.

**Comment by liggitt** [comment]
I think this is resolved in master and will be resolved in 1.27.2 by https://github.com/kubernetes/kubernetes/pull/117752

until 1.27.2 is available, setting the NewVolumeManagerReconstruction=false feature gate in the kubelet should resolve this

**Comment by SergeyKanzhelev** [comment]
@discostur can you please check if this is resolved for you?

/triage needs-information
